### PR TITLE
update microsoft.com/net links

### PR DESCRIPTION
**Pull request by mairaw**
Better to stop using microsoft.com/net links and use the actual target since the dot.net site no longer uses those

For Linux instructions, I changed to use the docs ones

### CODE REVIEW

**Comment by mairaw** [comment]
@leecow @rbhanda can one of you review this PR?

File: release-notes/1.0/1.0.1-sdk-release-notes.md [hunk 1]
```diff
@@ -6,15 +6,15 @@
 * [.NET Core 1.0.4](https://github.com/dotnet/core/blob/master/release-notes/1.0/1.0.4.md)
 * [.NET Core 1.1.1](https://github.com/dotnet/core/blob/master/release-notes/1.1/1.1.1.md)
 
-[.NET Core 1.0.4](https://github.com/dotnet/core/blob/master/release-notes/1.0/1.0.4.md) was released today. Both releases include reliability updates to improve the quality of .NET Core. You can download the .NET Core Runtime releases via our [.NET Core Runtimes download page](https://www.microsoft.com/net/download/core#/runtime). If you are looking for the .NET Core SDK, to get the latest tools, try the [.NET Core SDK download page](https://www.microsoft.com/net/download/core#/sdk).
+[.NET Core 1.0.4](https://github.com/dotnet/core/blob/master/release-notes/1.0/1.0.4.md) was released today. Both releases include reliability updates to improve the quality of .NET Core. You can download the .NET Core Runtime releases via our [.NET Core Runtimes download page](https://dotnet.microsoft.com/download/dotnet-core/current/runtime). If you are looking for the .NET Core SDK, to get the latest tools, try the [.NET Core SDK download page](https://dotnet.microsoft.com/download).
```
Comment (leecow) [inline release-notes/1.0/1.0.1-sdk-release-notes.md:9]: dotnet-core/current/runtime only offers Windows and Linux. Is that intentional?

File: release-notes/2.1/2.1.10/2.1.10-download.md [hunk 2]
```diff
@@ -155,7 +155,7 @@ This will install the ASP.NET Core Module for IIS.
 [checksums-runtime]: https://dotnetcli.blob.core.windows.net/dotnet/checksums/2.1.10-runtime-sha.txt
 [checksums-sdk]: https://dotnetcli.blob.core.windows.net/dotnet/checksums/2.1.506-sdk-sha.txt
 
-[linux-install]: https://www.microsoft.com/net/download/linux
+[linux-install]: https://docs.microsoft.com/dotnet/core/install/linux
```
Comment (leecow) [inline release-notes/2.1/2.1.10/2.1.10-download.md:158]: This is just the docs page whereas the previous link is the actual download.

File: release-notes/1.0/1.0.1-sdk-release-notes.md (hunk 1 above)
Comment (mairaw) [reply release-notes/1.0/1.0.1-sdk-release-notes.md:9]: I think it was done that way intentionally in the past but we're adding macOS to that page on a PR I'm working on

**Review by leecow — APPROVED** [review]

**Comment by rbhanda** [comment]
LGTM

**Review by rbhanda — APPROVED** [review]

**Pull request merged**